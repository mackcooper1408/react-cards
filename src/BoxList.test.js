import React from "react";
import { render, fireEvent } from "@testing-library/react";

import BoxList from "./BoxList";

it("renders without crashing", function () {
  render(<BoxList />);
});


it("matches snapshot", function () {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});


it("can add a new item", function () {
  const { getByLabelText, queryByText, container, debug } = render(<BoxList />);

  // no items yet
  expect(container.querySelector(".Box")).not.toBeInTheDocument();
  // expect(queryByText("ice cream: 100")).not.toBeInTheDocument();

  const heightInput = getByLabelText("Height:");
  const widthInput = getByLabelText("Width:");
  const bgInput = getByLabelText("Background Color:");
  const submitBtn = queryByText("Add a new box!");

  // fill out the form
  fireEvent.change(heightInput, {target: { value: 100 } });
  fireEvent.change(widthInput, {target: { value: 100 } });
  fireEvent.change(bgInput, {target: { value: "blue" } });
  fireEvent.click(submitBtn);

  // grab new box
  const box = container.querySelector(".Box");
  const styleWidth = box.style.width;
  const styleHeight = box.style.height;
  const styleBackgroundColor = box.style["background-color"];
  console.log("style ---> ", styleWidth);
  debug(box);
  // expect box to be on page
  expect(box).toBeInTheDocument();
  expect(styleHeight).toEqual("100px");
  expect(styleWidth).toEqual("100px");
  expect(styleBackgroundColor).toEqual("blue");

});
