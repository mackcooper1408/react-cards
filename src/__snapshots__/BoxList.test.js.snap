// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form>
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        type="number"
        value=""
      />
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        type="number"
        value=""
      />
      <label
        for="bgColor"
      >
        Background Color:
      </label>
      <input
        id="bgColor"
        name="bgColor"
        value=""
      />
      <button>
        Add a new box!
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
